rust/feature-flags/src/flag_matching.rs

```diff
@@ -147,20 +239,27 @@ impl FeatureFlagMatcher {
         Ok(self.check_rollout(feature_flag, rollout_percentage))
     }
 
-    async fn get_target_properties(
-        &mut self,
+    async fn get_group_properties(
+        &self,
         feature_flag: &FeatureFlag,
-        properties: &[PropertyFilter],
     ) -> Result<HashMap<String, Value>, FlagError> {
-        self.get_person_properties(feature_flag.team_id, properties)
-            .await
-        // TODO handle group properties, will go something like this
-        // if let Some(group_index) = feature_flag.get_group_type_index() {
-        //     self.get_group_properties(feature_flag.team_id, group_index, properties)
-        // } else {
-        //     self.get_person_properties(feature_flag.team_id, properties)
-        //         .await
-        // }
+        let group_index = match feature_flag.get_group_type_index() {
+            Some(index) => index,
+            None => return Ok(HashMap::new()), // No group index, return empty HashMap
+        };
+
+        let index_to_name = self.cache.group_type_index_to_name().await?;
+
+        let group_name = match index_to_name.get(&group_index) {
+            Some(name) => name,
+            None => return Ok(HashMap::new()), // No group name found, return empty HashMap
+        };
+
+        Ok(self
+            .group_property_overrides
+            .as_ref()
+            .and_then(|overrides| overrides.get(group_name).cloned())
+            .unwrap_or_else(HashMap::new))
```

Comment:
@neilkakkar continuing my thought from above, is there any case where the flag has a group index but this method returns empty HashMap, or is the idea that if I ever have a flag with a group property aggregation index but no props on the payload, I should always go fetch them from the DB (I haven't written that method yet).

My assumptions is that if the group properties condition exists on the flag, but there are no group properties in the payload that obey these conditions, I should go fetch the properties.

Just wanted to confirm my understanding!

Reply:
you only need to fetch the props if the flag in question depends on group properties, but yeah mostly you have this right

Reply:
> you only need to fetch the props if the flag in question depends on group properties, but yeah mostly you have this right

and we know this if the flag has an `aggregation_group_type_index`, right?

Reply:
aggregation group type index tells us its a group flag, so its necessary but not sufficient. It's analogous to person flags: You fetch properties if there's some `conditions` on the flag that say property key = value. If there's no property, it's a simple flag with only a rollout %, no properties needed.